### PR TITLE
Application Gateway WAF Rate Limit feature swagger changes

### PR DESCRIPTION
**Pull request by sraghavan-msft**
# Application Gateway WAF Rate Limit feature changes

PR captures configuration model changes to support Application Gateway WAF Rate Limit feature. This feature is built on the existing Custom Rule framework. A new Custom Rule type "RateLimitRule" has been added. Along with this, a new field GroupByUserSession has been added to allow users to configure how to identify unique user sessions or to rate limit requests to an url endpoint by choosing "None"

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2022-11-01/examples/WafListAllPolicies.json [hunk 1]
```diff
@@ -87,6 +87,51 @@
...
+                  "priority": 3,
+                  "rateLimitDuration": "OneMin",
+                  "rateLimitThreshold": 10,
+                  "ruleType": "RateLimitRule",
+                  "matchConditions": [
+                    {
+                      "matchVariables": [
+                        {
+                          "variableName": "RemoteAddr",
+                          "selector": null
```
Comment (tejasshah7) [inline specification/network/resource-manager/Microsoft.Network/stable/2022-11-01/examples/WafListAllPolicies.json:102]: Will this come back if it is null?

**Comment by raych1** [comment]
@sraghavan-msft , please fix `lintDiff` errors.

File: specification/network/resource-manager/Microsoft.Network/stable/2022-11-01/examples/WafListAllPolicies.json (hunk 1 above)
Comment (sraghavan-msft) [reply specification/network/resource-manager/Microsoft.Network/stable/2022-11-01/examples/WafListAllPolicies.json:102]: I assume you meant come back from NRP k/v store. If so, the answer is no.

Comment (sraghavan-msft) [reply specification/network/resource-manager/Microsoft.Network/stable/2022-11-01/examples/WafListAllPolicies.json:102]: Just realized that the "selector" your were pointing to is under "matchConditions". This section I copied over as is from the previous custom rules in the examples. They all have "selector": null in them. Do you want me to remove this across all examples? I have updated the examples such that the rate limit custom rule have match conditions that make sense within the context of each example.

**Comment by sraghavan-msft** [comment]
Pasting this comment from #23022 as it applies to this PR as well

@lirenhe The Breaking Change failure in this PR is due to a change that was merged directly in the main branch by Appgw team. It is not related to the changes in this PR. Below is a PR that was approved and merged with the same failure for your reference.

cc @tejasshah7 @MikhailTryakhov

#22457

**Review by raych1 — APPROVED** [review]

**Pull request merged**